generators/messaging-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/MessageListenerCodeGenerator.java

```diff
@@ -157,7 +156,7 @@ private AnnotationSpec generateMessageDrivenAnnotation(final Component component
                         generateActivationConfigPropertyAnnotation(DESTINATION_TYPE, component.inputDestinationType().getName()))
                 .addMember(ACTIVATION_CONFIG_PARAMETER, "$L",
                         generateActivationConfigPropertyAnnotation(DESTINATION_LOOKUP, resourceUri.destinationName()));
-        if (!containsGeneralJsonMimeType(actions)) {
+        if (!containsGeneralJsonMimeType(actions) && baseUri.component() != EVENT_LISTENER) {
```

Comment:
Might be worth pulling out to isNotEventListener() method, and adding a comment WHY this is important (for future maintenance)